src/main/java/seedu/address/MainApp.java

```diff
@@ -89,7 +94,23 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
                     + " Will be starting with an empty AddressBook.");
             initialData = new AddressBook();
         }
-        return new ModelManager(initialData, userPrefs);
+
+        Optional<ReadOnlyCalendar> calendarOptional;
+        ReadOnlyCalendar initialCalendarData;
+        try {
+            calendarOptional = storage.readCalendar();
+            if (!calendarOptional.isPresent()) {
+                logger.info("Creating a new data file " + storage.getCalendarFilePath()
+                        + " populated with a sample Calendar.");
+            }
+            initialCalendarData = calendarOptional.orElseGet(SampleDataUtil::getSampleCalendar);
```

Comment:
Not too sure but might be possible to do something like this:
`calendarOptional = storage.readCalendar().orElse(logger.info("Creating a new data file " + storage.getCalendarFilePath()
                        + " populated with a sample Calendar."));`

Reply:
Then logger.info (...) would be returned and stored into the optional calendar variable. I don't think that works. But it might be better to write it as optionalCalendar.ifPresent(calander -> logger.info....)

Reply:
But it's a bit odd because the calendar object is not utilised in the consumer